synapse/app/_base.py

```diff
@@ -266,9 +268,37 @@ def handle_sighup(*args, **kwargs):
         # It is now safe to start your Synapse.
         hs.start_listening(listeners)
         hs.get_datastore().start_profiling()
+
+        setup_sentry(hs)
```

Comment:
doesn't this want to be earlier?

Reply:
I'd expect any errors to happen before then to be fatal configuration errors, which I don't think we'd want to bother reporting to sentry, so I don't think it really matters too much where in the setup code we initialise it.